elevate-self-creation-portal/controllers/selfcreationportal.js

```diff
@@ -1,8 +1,167 @@
 const routeConfigs = require('../constants/routes')
 const requesters = require('../utils/requester')
-const requestParser = require('../utils/requestParser')
+const common = require('../constants/common')
+
+const fetchResources = async (req, res, selectedConfig) => {
+	try {
+
+		const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
+		let projectPath = null
+		let projectBaseUrl = null
+		let surveyPath = null
+		let surveyBaseUrl = null
+
+		selectedConfig?.targetRoute?.paths.map((targetRoute) => {
+			if(targetRoute.path == common.PROJECT_END_POINT){
+				projectPath = targetRoute.path
+				projectBaseUrl = targetRoute.baseUrl
+			}
+			if(targetRoute.path == common.SURVEY_END_POINT){
+				surveyPath = targetRoute.path
+				surveyBaseUrl =targetRoute.baseUrl
+			}
+		})
+
+		// fetch the max limit from the env file for the DB Find API
+		const max_limit = process.env.RESOURCE_MAX_FETCH_LIMIT ? parseInt(process.env.RESOURCE_MAX_FETCH_LIMIT, 10) : 1000
+
+		let response = { result: { data: [] } }
+
+		let proceedToCallProjectService = false
+		let proceedToCallSurveyService = false
+
+		let projectResponse = {}
+		let surveyResponse = {}
+
+		let projectReqBody = {
+			query: {}
+		}
+		let surveyReqBody = {
+			query: {}
+		}
+
+		let projectHeader = {}
+		let surveyHeader = {}
+		projectHeader[common.AUTH_TOKEN_KEY] = req.headers[common.AUTH_TOKEN_KEY].replace(/^(Bearer|bearer)\s*/, '')
```

Comment:
@adithyadinesh0412  can we do common heeader the code is repeating 
const headers = {
      [common.AUTH_TOKEN_KEY]: req.headers[common.AUTH_TOKEN_KEY].replace(/^(Bearer|bearer)\s*/, ''),
      [common.HEADER_CONTENT_TYPE]: req.headers[common.HEADER_CONTENT_TYPE] || 'application/json',
    };